File: node.js/src/slimmed/domain/books/factories/bookFactory.ts
import { State, Book } from '../book';
import {
  Author,
  BookId,
  Chapter,
  CommitteeApproval,
  Format,
  Genre,
  ISBN,
  Reviewer,
  Title,
  Translation,
} from '../entities';
import { IPublishingHouse } from '../services/publishingHouse';

export interface IBookFactory {
  create(
    bookId: BookId,
    state: State,
    title: Title,
    author: Author,
    publishingHouse: IPublishingHouse,
    genre: Genre | null,
    isbn: ISBN | null,
    committeeApproval: CommitteeApproval | null,
    reviewers: Reviewer[] | null,
    chapters: Chapter[] | null,
    translations: Translation[] | null,
    formats: Format[] | null,
  ): Book;
}

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/factories/BookFactory.java
package io.eventdriven.slimdownaggregates.original.domain.books.factories;

import io.eventdriven.slimdownaggregates.original.domain.books.Book;
import io.eventdriven.slimdownaggregates.original.domain.books.entities.*;
import io.eventdriven.slimdownaggregates.original.domain.books.services.PublishingHouse;
import io.eventdriven.slimdownaggregates.original.infrastructure.valueobjects.*;

import java.time.LocalDate;
import java.util.List;

public interface BookFactory {
  Book Create(
    BookId bookId,
    Book.State state,
    Title title,
    Author author,
    PublishingHouse publishingHouse,
    Publisher publisher,
    PositiveInt edition,
    Genre genre,
    ISBN isbn,
    LocalDate publicationDate,
    PositiveInt totalPages,
    PositiveInt numberOfIllustrations,
    NonEmptyString bindingType,
    NonEmptyString summary,
    CommitteeApproval committeeApproval,
    List<Reviewer> reviewers,
    List<Chapter> chapters,
    List<Translation> translations,
    List<Format> formats
  );
}
